src/game/Globals/ObjectMgr.cpp

```diff
@@ -3681,31 +3681,31 @@ HonorStandingList ObjectMgr::GetStandingListBySide(uint32 side)
     }
 }
 
-HonorStanding* ObjectMgr::GetHonorStandingByGUID(uint32 guid, uint32 side)
+HonorStanding ObjectMgr::GetHonorStandingByGUID(uint32 guid, uint32 side)
 {
     HonorStandingList standingList = sObjectMgr.GetStandingListBySide(side);
 
     for (HonorStandingList::iterator itr = standingList.begin(); itr != standingList.end() ; ++itr)
         if (itr->guid == guid)
-            return itr->GetInfo();
+            return *itr;
 
-    return 0;
+    return HonorStanding();
 }
 
 
-HonorStanding* ObjectMgr::GetHonorStandingByPosition(uint32 position, uint32 side)
+HonorStanding ObjectMgr::GetHonorStandingByPosition(uint32 position, uint32 side)
 {
     HonorStandingList standingList = sObjectMgr.GetStandingListBySide(side);
     uint32 pos = 1;
 
     for (HonorStandingList::iterator itr = standingList.begin(); itr != standingList.end() ; ++itr)
     {
         if (pos == position)
-            return itr->GetInfo();
+            return *itr;
         pos++;
     }
 
-    return 0;
+    return HonorStanding();
```

Comment:
Given the semantics of these functions, that are not guaranteed to give a result, maybe returning an ```std::optional<HonorStanding>``` would make sense in these cases? 

Given the already existing default constructor of HonorStanding, which constructs some kind of nothing-object, his is acceptable for me as well, but I'd personally prefer an API that makes it clear that "this may not return what you were looking for".

Reply:
Thank you. I did not know about std::optional and it looks like I can use that in more situations in the future!